### DISCUSSION TITLE
Backblaze B2 - Not Showing All Directories

### DISCUSSION THREAD

**Discussion by ghost** [iterate-ch/cyberduck#14759] [Q&A] [answered]
Cyberduck Version: 8.6.0

Cyberduck is not showing all my directories inside a bucket on Backblaze B2, it is only listing 3 directories but I have more than that. When I browse files on backblaze.com directly I can see everything fine. This issue appears to be a bug with cyberduck. Please fix it.

Please let me know if you need more information.

**Comment by dkocher** [comment, marked as answer]
We use a convention also used by the Backblaze web application that creates .bzEmpty 0KB files as directory placeholders as the B2 object storage has no concept of folders. When only a common prefix is found because of a file named like folder/file but no file folder/.bzEmpty is found, we assume the the folder has been deleted and files are no longer displayed by default. Using Cyberduck, the folder will show when you choose _View → Show Hidden Files_ in Cyberduck.

As a workaround to always display the folder, navigate into the folder and manually create the missing directory placeholder file `.bzEmpty` by choosing _File → New File…_ in Cyberduck.

**Reply by metadaddy** [reply]
Related: https://github.com/iterate-ch/cyberduck/issues/14768

**Reply by metadaddy** [reply]
Hi @dkocher - the issue here is that this is just a convention, used by the Backblaze web console because users want to be able to create a 'folder' (which doesn't really exist in the cloud object storage model) independently of uploading files. The vast majority of objects in Backblaze B2 are created by applications which do not follow this convention, because they don't need to. They just write to `folder/subfolder/file0001`.

The assumption that "the folder has been deleted" is incorrect. It's not uncommon for users to use Cyberduck to verify that an application, for example, their NAS backup, is working correctly. In Cyberduck 8.5.7, when I opened my Synology NAS backup bucket, I saw a "folder", `DS920plus_1.hbk`, that I could click into to see that there were files there. Now I see nothing.

The convenience of not seeing "folders" whose contents have been deleted is massively outweighed by the inconvenience of not seeing my "folders" that actually exist.
